build.sbt

```diff
@@ -1303,6 +1305,79 @@ lazy val `runtime-language-epb` =
       instrumentationSettings
     )
 
+/**
+ * Runs gu (GraalVM updater) command with given `args`.
+ * For example `runGu(Seq("install", "js"))`.
+ * @param logger Logger for the `gu` command.
+ * @param args Arguments for the `gu` command.
+ */
+def runGu(logger: ManagedLogger, args: Seq[String]): String = {
+  val javaHome = new File(
+    System.getProperty("java.home")
+  )
+  val os = {
+    if (Platform.isLinux) DistributionPackage.OS.Linux
+    else if (Platform.isMacOS) DistributionPackage.OS.MacOS
+    else if (Platform.isWindows) DistributionPackage.OS.Windows
+    else throw new RuntimeException("Unknown platform")
+  }
+  packageBuilder.gu(
+    logger,
+    os,
+    javaHome,
+    args:_*
+  )
+}
+
+def installedGuComponents(logger: ManagedLogger): Seq[String] = {
+  val componentList = runGu(
+    logger,
+    Seq("list")
+  )
+  val components = componentList
+    .linesIterator
+    .drop(2)
+    .map { line =>
+      line
+        .split(" ")
+        .head
+    }
+    .toList
+  logger.debug(s"Installed GU components = $components")
+  if (!components.contains("graalvm")) {
+    throw new RuntimeException(s"graalvm components is not in $components")
+  }
+  components
+}
+
+lazy val installGraalJs = taskKey[Unit]("Install graaljs GraalVM component")
+ThisBuild / installGraalJs := {
+  val logger = streams.value.log
+  if (!installedGuComponents(logger).contains("js")) {
+    logger.info("Installing js GraalVM component")
+    runGu(
+      logger,
+      Seq("install", "js")
+    )
+  }
+}
+
+lazy val installNativeImage = taskKey[Unit]("Install native-image GraalVM component")
```

Comment:
This logic already is present in the RRust build script, if you use it to enter `sbt` shell then you'll have native-image installed.

Reply:
I have just recently noticed that. Nevertheless we still need `installGraalJs`, as it is not shipped by default in the latest GraalVM release. Moreover, this task does not install anything, if `native-image` is already installed. The same is true for `installGraalJs`.  Besides, I don't know how other engine devs, but I don't usually use the rust build script, I just use `sbt`, which makes these two tasks very convenient.